extmod/network_nina.c

```diff
@@ -0,0 +1,591 @@
+        return -1;
+    }
+    return ret;
+}
+
+static int py_nina_socket_setsockopt(mod_network_socket_obj_t *socket, mp_uint_t
+    level, mp_uint_t opt, const void *optval, mp_uint_t optlen, int *_errno) {
+    int ret = nina_socket_setsockopt(socket->fileno, level, opt, optval, optlen);
+    if (ret < 0) {
+        *_errno = ret;
+        py_nina_socket_close(socket);
+        return -1;
+    }
+    return 0;
+}
+
+static int py_nina_socket_settimeout(mod_network_socket_obj_t *socket, mp_uint_t timeout_ms, int *_errno) {
+    if (timeout_ms == UINT32_MAX) {
+        // no timeout is given, set the socket to blocking mode.
+        timeout_ms = 0;
+    }
+    socket->timeout = timeout_ms;
+    return 0;
+}
+
+static int py_nina_socket_ioctl(mod_network_socket_obj_t *socket, mp_uint_t request, mp_uint_t arg, int *_errno) {
+    *_errno = MP_EIO;
+    return -1;
+}
+
+static MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(py_nina_active_obj, 1, 2, py_nina_active);
+static MP_DEFINE_CONST_FUN_OBJ_KW(py_nina_connect_obj, 1,   py_nina_connect);
+static MP_DEFINE_CONST_FUN_OBJ_1(py_nina_disconnect_obj,    py_nina_disconnect);
+static MP_DEFINE_CONST_FUN_OBJ_1(py_nina_isconnected_obj,   py_nina_isconnected);
+static MP_DEFINE_CONST_FUN_OBJ_1(py_nina_connected_sta_obj, py_nina_connected_sta);
+static MP_DEFINE_CONST_FUN_OBJ_2(py_nina_wait_for_sta_obj,  py_nina_wait_for_sta);
+static MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(py_nina_ifconfig_obj, 1, 2, py_nina_ifconfig);
+static MP_DEFINE_CONST_FUN_OBJ_1(py_nina_netinfo_obj,       py_nina_netinfo);
+static MP_DEFINE_CONST_FUN_OBJ_1(py_nina_scan_obj,          py_nina_scan);
+static MP_DEFINE_CONST_FUN_OBJ_1(py_nina_get_rssi_obj,      py_nina_get_rssi);
+static MP_DEFINE_CONST_FUN_OBJ_1(py_nina_fw_version_obj,    py_nina_fw_version);
+
+static const mp_rom_map_elem_t nina_locals_dict_table[] = {
+    { MP_ROM_QSTR(MP_QSTR_active),              MP_ROM_PTR(&py_nina_active_obj) },
+    { MP_ROM_QSTR(MP_QSTR_connect),             MP_ROM_PTR(&py_nina_connect_obj) },
+    { MP_ROM_QSTR(MP_QSTR_config),              MP_ROM_PTR(&py_nina_connect_obj) },
+    { MP_ROM_QSTR(MP_QSTR_start_ap),            MP_ROM_PTR(&py_nina_connect_obj) },
+    { MP_ROM_QSTR(MP_QSTR_disconnect),          MP_ROM_PTR(&py_nina_disconnect_obj) },
+    { MP_ROM_QSTR(MP_QSTR_isconnected),         MP_ROM_PTR(&py_nina_isconnected_obj) },
+    { MP_ROM_QSTR(MP_QSTR_connected_sta),       MP_ROM_PTR(&py_nina_connected_sta_obj) },
+    { MP_ROM_QSTR(MP_QSTR_wait_for_sta),        MP_ROM_PTR(&py_nina_wait_for_sta_obj) },
+    { MP_ROM_QSTR(MP_QSTR_ifconfig),            MP_ROM_PTR(&py_nina_ifconfig_obj) },
+    { MP_ROM_QSTR(MP_QSTR_netinfo),             MP_ROM_PTR(&py_nina_netinfo_obj) },
+    { MP_ROM_QSTR(MP_QSTR_scan),                MP_ROM_PTR(&py_nina_scan_obj) },
+    { MP_ROM_QSTR(MP_QSTR_rssi),                MP_ROM_PTR(&py_nina_get_rssi_obj) },
+    { MP_ROM_QSTR(MP_QSTR_fw_version),          MP_ROM_PTR(&py_nina_fw_version_obj) },
```

Comment:
`fw_version()` would be better as `config("fw_version")`, and `rssi()` should be `status("rssi")`.